cmd/juju/application/bundle/bundle.go

```diff
@@ -319,7 +299,31 @@ func gatherErrors(ds BundleDataSource) []error {
 	return returnErrors
 }
 
-func verifyBundle(data *charm.BundleData, bundleDir string, verifyConstraints, verifyStorage, verifyDevices func(string) error) error {
+func verifyBaseBundle(base BundleDataSource) error {
+	verifyBaseConstraints := func(s string) error {
+		bundleConstraints, err := constraints.Parse(s)
+		if err != nil {
+			return err
+		}
+		if bundleConstraints.HasImageID() {
+			return errors.NotSupportedf("'image-id' constraint in a base bundle")
+		}
+		return nil
+	}
+
+	return verifyBundle(base.Parts()[0].Data, base.BasePath(), verifyBaseConstraints)
```

Comment:
@hmlanigan @anvial is this always true (`(BundleDataSource).Parts[0]` equals to the base bundle)?

Reply:
Imho, yes